mesonbuild/environment.py

```diff
@@ -747,7 +747,10 @@ def _set_default_binaries_from_env(self) -> None:
         for (name, evar), for_machine in itertools.product(opts, MachineChoice):
             p_env = _get_env_var(for_machine, self.is_cross_build(), evar)
             if p_env is not None:
-                self.binaries[for_machine].binaries.setdefault(name, mesonlib.split_args(p_env))
+                if os.path.exists(p_env):
+                    self.binaries[for_machine].binaries.setdefault(name, [p_env])
+                else:
+                    self.binaries[for_machine].binaries.setdefault(name, mesonlib.split_args(p_env))
```

Comment:
Wondering if there are other places we should be doing the same thing. I see for example scanbuild in that same file.

Reply:
`detect_scanbuild` does not use `os.path.join`, rather `shutil.which`. Only paths that will be passed to compilers need to be converted.